docs/reference/nuctl/nuctl.md

```diff
@@ -1,48 +1,63 @@
-- [Download](#download)
+- [Overview](#overview)
+- [Download and Installation](#download-n-install)
 - [Usage](#usage)
-- [Running platform](#running-platform)
-- [Docker](#docker)
-- [Kubernetes](#kubernetes)
+- [The running platform](#running-platform)
+  - [Docker](#docker)
+  - [Kubernetes](#kubernetes)
 
-## About
+<a id="overview"></a>
+## Overview
 
-Nuctl is Nuclio's command-line interface that provides you with all the nuclio features, accessible from your terminal
+`nuctl` is Nuclio's command-line interface (CLI), which provides command-line access to Nuclio features.
 
-## Download
+<a id="download-n-install"></a>
+## Download and Installation
 
-To install Nuctl all you need to do, is simply visit Nuclio [releases page](https://github.com/nuclio/nuclio/releases)
-and download the binary appropriate to your platform (e.g.: `darwin` if you have `macOS`)
+To install `nuctl`, simply visit the Nuclio [releases page](https://github.com/nuclio/nuclio/releases) and download the appropriate CLI binary for your environment (for example, **nuctl-&lt;version&gt;-darwin-amd64** for a machine running macOS).
 
-You can use the following bash snippet to download the latest nuctl release
-```bash
+You can use the following command to download the latest `nuctl` release:
+```sh
 curl -s https://api.github.com/repos/nuclio/nuclio/releases/latest \
 			| grep -i "browser_download_url.*nuctl.*$(uname)" \
 			| cut -d : -f 2,3 \
 			| tr -d \" \
 			| wget -O nuctl -qi - && chmod +x nuctl
 ```
 
+<a id="usage"></a>
 ## Usage
 
-Once downloaded, an informative help section is available using `nuctl --help`
+After you download `nuctl`, use the `--help` option to see full usage instructions:
+```sh
+nuctl --help
+```
 
-## Running platform
+<a id="running-platform"></a>
+## The running platform
 
-Nuctl will automatically identify its running platform, whether it is Docker or Kubernetes.
+`nuctl` automatically identifies the platform from which it's being run.
+You can also use the `--platform` CLI flag to ensure that you're running against a specific platform, as explained in this reference:
 
-To ensure you run Nuctl against a specific platform, use `--platform kube` for Kubernetes or `--platform local` for docker
+- [Docker](#docker)
+- [Kubernetes](#kubernetes)
 
+<a id="docker"></a>
 ### Docker
 
-An example of function deployment using Nuctl against Docker can be found [here](/docs/setup/docker/getting-started-docker.md)
+To ensure that you're running `nuctl` on Docker, add `--platform local` to the CLI command.
+
+For an example of function deployment using `nuctl` against Docker, see the Nuclio [Docker getting-started guide](/docs/setup/docker/getting-started-docker.md).
 
+<a id="kubernetes"></a>
 ### Kubernetes
 
-When running in Kubernetes, Nuctl would require you running a registry on your Kubernetes cluster and access to a `kubeconfig`
+To ensure that you're running `nuctl` on Kubernetes, add `--platform kube` to the CLI command.
```

Comment:
```suggestion
To force `nuctl` to work against a  Kubernetes `nuclio` instance, add `--platform kube` to the CLI command.
```
wrt "ensuring" here and above.

It sounds a bit off to me. it should be "force" IMO.
The reality is, that nuctl on the same machine can work both against a local docker nuclio installation, and a remote k8s installation alternatively. In this case the user will *have* to give one of the flags in some scenario because nuctl can't guess against which nuclio/platform it's supposed to work, and on the same machine will always work against either local / or k8s depending on it's auto detection, it may be necessary for the user to alternate between them if he/she so wishes
This is why I think "ensuring" is a bit misleading here - it's not just "making sure" - sometimes we have to TELL nuctl who to work against or it won't know / guess wrong.
The logic to auto detect a platform is this:
- if kubeconfig available at the default dir, or an alternative path to kubeconfig is passed in config or via env var - it's k8s, try to work against the k8s system using this kubeconfig, else - default to local (docker)

So, you see, the logic of auto detection might cause nuclio to always work against k8s, and ignore a local nuclio installation, which might be what the user wants

Reply:
@omesser this is a technical discussion that I'm not equipped to weight in. The "ensure" terminology was in the original version of the file (which I thought you had reviewed, as you approved PR #1708 ...). I don't mind replacing the two "To ensure that you're running `nucltl` on ..." (for Docker and k8s) with "To force `nuctl` to work with a ..." (I don't really like the "work against" terminology, especially as "work against" in English means to oppose ...). Also, based on Your previous comment regarding a local Dcoker, I would change the "Docker" references to "a local Docker registry". => @omesser @liranbg please approve before I make the changes.

Reply:
- "Docker ->  a local Docker registry" - That's a ❌  runtime != registry
- against vs on - when on docker/local runtime, nuctl must run on the same machine. With k8s, running "against" this cluster is usually (but not exclusively) done from outside of the cluster. Hence the more generic terminology. it is a tool communicating with a k8s cluster or a host running docker (locally).
- Unless got a semantically equivalent suggestion to denote a client-server like relationship, please change to:
```
To force `nuctl` to work against a  Kubernetes `nuclio` instance, add `--platform kube` to the CLI command.
```

Here, and in the local section:
```
To force `nuctl` to work locally, using docker daemon, add `--platform local` to the CLI command.
```

Reply:
@omesser @liranbg I rephrased a bit differently (commit 01abc11); please approve:
- local —
    ```markdown
    To force `nuctl` to run locally, using a Docker daemon, add `--platform local` to the CLI command.
    ```
- k8s —
    ```markdown
    To force `nuctl` to run against a Kubernetes instance of Nuclio, add `--platform kube` to the CLI command.
    ```

Reply:
Great, thanks!